File: packages/shared/mongodb/Conversation.ts
import type {
  ConversationMode,
  ConversationStatus,
  ISODateString,
  MessageRole,
  UUID,
} from "./types";

export interface ToolCall {
  tool: string;
  input: Record<string, unknown>;
  output_summary: string;
  at: ISODateString;
}

export interface ConversationMessage {
  role: MessageRole;
  content: string;
  at: ISODateString;
  tool_calls: ToolCall[] | null;
}

export interface Conversation {
  _id: UUID;
  updated_at: ISODateString | null;
  user_id: UUID;
  mode: ConversationMode;
  claim_id: UUID | null;
  title: string;
  messages: ConversationMessage[];
  trace_ids: string[];
  status: ConversationStatus;
  created_at: ISODateString;
  last_message_at: ISODateString;
  archived_at: ISODateString | null;
  agent_session_id?: string | null;
}

File: packages/shared/mongodb/conversation.py
"""Conversation collection — mirror of Conversation.ts."""

from typing import Any

from pydantic import BaseModel

from .base import BaseDocument
from .enums import ConversationMode, ConversationStatus, MessageRole


class ToolCall(BaseModel):
    tool: str
    input: dict[str, Any]
    output_summary: str
    at: str


class ConversationMessage(BaseModel):
    role: MessageRole
    content: str
    at: str
    tool_calls: list[ToolCall] | None


class Conversation(BaseDocument):
    user_id: str
    mode: ConversationMode
    claim_id: str | None
    title: str
    messages: list[ConversationMessage]
    trace_ids: list[str]
    status: ConversationStatus
    created_at: str
    last_message_at: str
    archived_at: str | None
